cli/lsp/documents.rs

```diff
@@ -1528,7 +1557,7 @@ impl Documents {
       reqs
     });
     self.dirty = false;
-    file_system_docs.dirty = false;
+    self.file_system_docs.set_dirty(false);
```

Comment:
If something else comes along while this is running and marks it as dirty, then this will change it as not dirty. Is that a concern? Maybe instead it should swap the value and mark it as not dirty above? Something like:

```
let is_fs_docs_dirty = self.file_system_docs.set_dirty(false); // make sure this is always done
if is_fs_docs_dirty && !self.dirty {
```